filters/src/passthrough.cpp

```diff
@@ -273,6 +273,106 @@ pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (PCLPointCloud2 &output)
+  {
+    // Only filter for non-finite entries then
+    for (const auto ii : indices)  // ii = input index
+    {
+      float pt[3];
+      memcpy (&pt[0], &input_->data[ii * input_->point_step + xyz_offset[0]], sizeof(float));
+      memcpy (&pt[1], &input_->data[ii * input_->point_step + xyz_offset[1]], sizeof(float));
+      memcpy (&pt[2], &input_->data[ii * input_->point_step + xyz_offset[2]], sizeof(float));
+      // Non-finite entries are always passed to removed indices
+      if (!std::isfinite (pt[0]) ||
+          !std::isfinite (pt[1]) ||
+          !std::isfinite (pt[2]))
+      {
+        if (extract_removed_indices_)
+          (*removed_indices_)[rii++] = ii;
+        continue;
+      }
+      indices[oii++] = ii;
+    }
+  }
+  else
+  {
+    // Attempt to get the field name's index
+    int distance_idx = pcl::getFieldIndex (*input_, filter_field_name_);
+    if (distance_idx == -1)
+    {
+      PCL_WARN ("[pcl::%s::applyFilter] Unable to find field name in point type.\n", getClassName ().c_str ());
+      indices.clear ();
+      removed_indices_->clear ();
+      return;
+    }
+
+    // Filter for non-finite entries and the specified field limits
+    for (const auto ii : indices)  // ii = input index
+    {
+      float pt[3];
+      memcpy (&pt[0], &input_->data[ii * input_->point_step + xyz_offset[0]], sizeof(float));
+      memcpy (&pt[1], &input_->data[ii * input_->point_step + xyz_offset[1]], sizeof(float));
+      memcpy (&pt[2], &input_->data[ii * input_->point_step + xyz_offset[2]], sizeof(float));
+      // Non-finite entries are always passed to removed indices
+      if (!std::isfinite (pt[0]) ||
+          !std::isfinite (pt[1]) ||
+          !std::isfinite (pt[2]))
+      {
+        if (extract_removed_indices_)
+          (*removed_indices_)[rii++] = ii;
+        continue;
+      }
+
+      // Get the field's value
+      const std::uint8_t* pt_data = reinterpret_cast<const std::uint8_t*> (&input_->data[ii * input_->point_step]);
+      float field_value = 0;
+      memcpy (&field_value, pt_data + input_->fields[distance_idx].offset, sizeof (float));
+
+      // Remove NAN/INF/-INF values. We expect passthrough to output clean valid data.
+      if (!std::isfinite (field_value))
+      {
+        if (extract_removed_indices_)
+          (*removed_indices_)[rii++] = ii;
+        continue;
+      }
+
+      // Outside of the field limits are passed to removed indices
+      if (!negative_ && (field_value < filter_limit_min_ || field_value > filter_limit_max_))
+      {
+        if (extract_removed_indices_)
+          (*removed_indices_)[rii++] = ii;
+        continue;
+      }
+
+      // Inside of the field limits are passed to removed indices if negative was set
+      if (negative_ && field_value >= filter_limit_min_ && field_value <= filter_limit_max_)
+      {
+        if (extract_removed_indices_)
+          (*removed_indices_)[rii++] = ii;
+        continue;
+      }
```

Comment:
These three conditions can be chained in a single `if`. Make use of indentation, new lines and comments to keep it as clear as you did here.

Reply:
Ok :+1: Should I do that [in the templated version](https://github.com/mvieth/pcl/blob/master/filters/include/pcl/filters/impl/passthrough.hpp#L130) as well?

Reply:
No. Let's leave that for a future PR with the aim of consolidating all variants of the filter over its different overloads.